### ISSUE TITLE
[BUG] Changing the keycloak root user password results in failed redeployment

### ISSUE THREAD

**Issue by iameskild** [Quansight/qhub#1165]
### OS system and architecture in which you are running QHub

macOS / Github Ubuntu runner

### Expected behavior

In our docs, we strongly encourage qhub admins to change their `initial_root_password` however doing so and redeploying results in a failed deployment.

### Actual behavior

The check for stage `05_kubernetes_keycloak` needs to be relaxed so that we can allow qhub admins to change their root user password without the deployment failing.

### How to Reproduce the problem?

Change root user password and redeploy qhub.

### Command output

```bash session
INFO:qhub.provider.terraform:terraform=/tmp/terraform/1.0.5/terraform output directory=stages/05-kubernetes-keycloak
INFO:qhub.provider.terraform:terraform output took 0.936 [s]
/opt/hostedtoolcache/Python/3.8.12/x64/lib/python3.8/site-packages/urllib3/connectionpool.py:1043: InsecureRequestWarning: Unverified HTTPS request is being made to host 'quansight-beta.qhub.dev'. Adding certificate verification is strongly advised. See: https://urllib3.readthedocs.io/en/1.26.x/advanced-usage.html#ssl-warnings
  warnings.warn(
/opt/hostedtoolcache/Python/3.8.12/x64/lib/python3.8/site-packages/urllib3/connectionpool.py:1043: InsecureRequestWarning: Unverified HTTPS request is being made to host 'quansight-beta.qhub.dev'. Adding certificate verification is strongly advised. See: https://urllib3.readthedocs.io/en/1.26.x/advanced-usage.html#ssl-warnings
  warnings.warn(
/opt/hostedtoolcache/Python/3.8.12/x64/lib/python3.8/site-packages/urllib3/connectionpool.py:1043: InsecureRequestWarning: Unverified HTTPS request is being made to host 'quansight-beta.qhub.dev'. Adding certificate verification is strongly advised. See: https://urllib3.readthedocs.io/en/1.26.x/advanced-usage.html#ssl-warnings
  warnings.warn(
/opt/hostedtoolcache/Python/3.8.12/x64/lib/python3.8/site-packages/urllib3/connectionpool.py:1043: InsecureRequestWarning: Unverified HTTPS request is being made to host 'quansight-beta.qhub.dev'. Adding certificate verification is strongly advised. See: https://urllib3.readthedocs.io/en/1.26.x/advanced-usage.html#ssl-warnings
  warnings.warn(
/opt/hostedtoolcache/Python/3.8.12/x64/lib/python3.8/site-packages/urllib3/connectionpool.py:1043: InsecureRequestWarning: Unverified HTTPS request is being made to host 'quansight-beta.qhub.dev'. Adding certificate verification is strongly advised. See: https://urllib3.readthedocs.io/en/1.26.x/advanced-usage.html#ssl-warnings
  warnings.warn(
/opt/hostedtoolcache/Python/3.8.12/x64/lib/python3.8/site-packages/urllib3/connectionpool.py:1043: InsecureRequestWarning: Unverified HTTPS request is being made to host 'quansight-beta.qhub.dev'. Adding certificate verification is strongly advised. See: https://urllib3.readthedocs.io/en/1.26.x/advanced-usage.html#ssl-warnings
  warnings.warn(
/opt/hostedtoolcache/Python/3.8.12/x64/lib/python3.8/site-packages/urllib3/connectionpool.py:1043: InsecureRequestWarning: Unverified HTTPS request is being made to host 'quansight-beta.qhub.dev'. Adding certificate verification is strongly advised. See: https://urllib3.readthedocs.io/en/1.26.x/advanced-usage.html#ssl-warnings
  warnings.warn(
/opt/hostedtoolcache/Python/3.8.12/x64/lib/python3.8/site-packages/urllib3/connectionpool.py:1043: InsecureRequestWarning: Unverified HTTPS request is being made to host 'quansight-beta.qhub.dev'. Adding certificate verification is strongly advised. See: https://urllib3.readthedocs.io/en/1.26.x/advanced-usage.html#ssl-warnings
  warnings.warn(
/opt/hostedtoolcache/Python/3.8.12/x64/lib/python3.8/site-packages/urllib3/connectionpool.py:1043: InsecureRequestWarning: Unverified HTTPS request is being made to host 'quansight-beta.qhub.dev'. Adding certificate verification is strongly advised. See: https://urllib3.readthedocs.io/en/1.26.x/advanced-usage.html#ssl-warnings
  warnings.warn(
/opt/hostedtoolcache/Python/3.8.12/x64/lib/python3.8/site-packages/urllib3/connectionpool.py:1043: InsecureRequestWarning: Unverified HTTPS request is being made to host 'quansight-beta.qhub.dev'. Adding certificate verification is strongly advised. See: https://urllib3.readthedocs.io/en/1.26.x/advanced-usage.html#ssl-warnings
  warnings.warn(
Attempt 1 failed connecting to keycloak master realm
Attempt 2 failed connecting to keycloak master realm
Attempt 3 failed connecting to keycloak master realm
Attempt 4 failed connecting to keycloak master realm
Attempt 5 failed connecting to keycloak master realm
Attempt 6 failed connecting to keycloak master realm
Attempt 7 failed connecting to keycloak master realm
Attempt 8 failed connecting to keycloak master realm
Attempt 9 failed connecting to keycloak master realm
Attempt 10 failed connecting to keycloak master realm
ERROR: unable to connect to keycloak master realm at url=https://quansight-beta.qhub.dev/auth/ with root credentials
```

### Versions and dependencies used.

qhub - main branch

### Compute environment

GCP

### Integrations

Keycloak

**Comment by viniciusdc** [comment]
Hi @iameskild, could you check if the `/tmp/QHUB_DEFAULT_PASSWORD` was modified as well? maybe some resources are still using the previous one, while the Keyclok provider is attempting a request with the new pass.

**Comment by danlester** [comment]
As briefly discussed yesterday, there is a "qhub-bot" password which is generated in-cluster by Terraform, and set with the Keycloak helm install.

In the file `qhub/template/stages/05-kubernetes-keycloak/modules/kubernetes/keycloak-helm/outputs.tf` we set `credentials` output for stage 05 of Terraform. This contains the root/initial-root-password username/password.

I think we should try changing this to qhub-bot/var.qhub-bot-password.

That would mean we are using the in-cluster password for qhub-bot when we do things like this in deploy.py:
```
        provision_05_kubernetes_keycloak(stage_outputs, config)

        with keycloak_provider_context(
            stage_outputs["stages/05-kubernetes-keycloak"]["keycloak_credentials"][
                "value"
            ]
        ):
            provision_06_kubernetes_keycloak_configuration(stage_outputs, config)
```

However, we then have to be careful where we're using keycloak_credentials elsewhere, e.g. on this line:
```
    print(
        f"Kubecloak master realm username={stage_outputs['stages/05-kubernetes-keycloak']['keycloak_credentials']['value']['username']} password=file://{os.path.join(tempfile.gettempdir(), 'QHUB_DEFAULT_PASSWORD')}"
    )
```

it would now show `qhub-bot` as the username when it used to show `root` (it should continue to show `root` to the user). Personally, I don't think it should be attempting to get the password from the file at all - that is **very 'wrong'** in the sense that the file would only have been set during `qhub init`. It would be more correct to provide the `initial_root_password` value from qhub-config.yaml, but of course that may be wrong if it was changed or removed after initial deployment.

Note that the `qhub keycloak` command should still only try the initial_root_password from qhub-config.yaml - that is not supposed to work long past initial deployment (and is not really encouraged for normal admins to use at all).
